### PR TITLE
Ports Pixel Shifting

### PR DESCRIPTION
**Pull request by Cerebulon**
CTRL + SHIFT + Movement Keys to scoot around juuuuuust a little.
https://github.com/VOREStation/VOREStation/pull/9265 + various fixes

:cl:
rscadd - Added pixel shift movement. Ctrl+shift+movement key to move around within a single tile.
/:cl:

### CODE REVIEW

**Comment by zekesulastin** [comment]
Aside from the CI failure, it also looks like you didn’t add CTRL+SHIFT+W/A/S/D for hotkey mode :(

**Comment by Cerebulon** [comment]
> 
> 
> Aside from the CI failure, it also looks like you didn’t add CTRL+SHIFT+W/A/S/D for hotkey mode :(

NORTH, SOUTH, EAST and WEST should reference the actual movement keys afaik?

**Comment by zekesulastin** [comment]
> 
> 
> > Aside from the CI failure, it also looks like you didn’t add CTRL+SHIFT+W/A/S/D for hotkey mode :(
> 
> NORTH, SOUTH, EAST and WEST should reference the actual movement keys afaik?

Nope - they are defined further down in the alphabetical section of the file

**Comment by MistakeNot4892** [comment]
Personally I really don't like this feature. It allows you to mess with offsets and interrupt animations for the sake of the minor RP benefit of being to pose yourself. Positioning on turfs can already be a bit confusing and I think this has the potential to make it worse.

Not blocking or anything, but won't be approving this personally. Sort of an abstaining review I guess?

**Comment by Atermonera** [comment]
This feels like it will introduce a lot of unintuitive behaviour where animations are interrupted or offsets reset unexpectedly. It might introduce weird behaviour in combat from projectile weapon targeting?

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -1009,6 +1008,38 @@
 	set hidden = 1
 	set_face_dir(client.client_dir(WEST))
 
+/mob/verb/shiftnorth()
+	set hidden = TRUE
+	if(!canface())
+		return FALSE
+	if(pixel_y <= (default_pixel_y + 16))
+		pixel_y++
+		is_shifted = TRUE
```
Comment (Atermonera) [inline code/modules/mob/mob.dm:1017]:
```suggestion
	if(custom_pixel_y < 16)
		pixel_y++
		custom_pixel_y++
```
And so on and so forth.

File: code/modules/mob/living/living_movement.dm [hunk 2]
```diff
@@ -210,6 +210,10 @@ default behaviour is:
 /mob/living/Moved(var/atom/oldloc, direct, forced, movetime)
 	. = ..()
 	handle_footstep(loc)
+	if(is_shifted)
+		is_shifted = FALSE
+		pixel_x = default_pixel_x
+		pixel_y = default_pixel_y
```
Comment (Atermonera) [inline code/modules/mob/living/living_movement.dm:216]: By tracking custom offsets independently, we just subtract pixel_x/y by those values and set the tracking vars to 0
```suggestion
if(custom_pixel_x)
	pixel_x -= custom_pixel_x
	custom_pixel_x = 0

if(custom_pixel_y)
	pixel_y -= custom_pixel_y
	custom_pixel_y = 0
```

File: code/modules/mob/mob.dm (hunk 1 above)
Comment (Atermonera) [reply code/modules/mob/mob.dm:1017]: Also, I'm guessing hotkey macros are why there's 4 separate procs here, and not one that just takes in a DIR?

**Review by Atermonera — CHANGES_REQUESTED** [review]

**Comment by Atermonera** [comment]
Stale, no plans to work on it due to opposition

**Comment by Spookerton** [comment]
@Atermonera reopen intended?

**Comment by Spookerton** [comment]
I will assume not, given a month.

As far as I'm concerned the opposition to this is entirely correct. Being able to do some little pixel shifting to the detriment of numerous other real mechanics is not a good trade.

**Pull request closed without merging**